src/kixi/datastore/communications/coreasync.clj

```diff
@@ -11,37 +11,49 @@
      :out (async/mult c)}))
 
 (defn tap-channel-with
-  [cm selector processor put-result]
-  (let [tapper (async/chan)]
+  [cm ns selector processor put-result]
+  (let [tapper (async/chan 5)]
     (async/go-loop [msg (async/<! tapper)]
       (when msg
         (try
           (when (selector msg)
-            (let [result (processor msg)]
-              (when put-result
-                (async/>! (:in cm)
-                          result))))
+            (async/go
+              (let [result (processor msg)]
+                (when (and put-result 
+                           result)
+                  
+                  (async/>! (:in cm)
+                            result)))))
           (catch Exception e
-            (error e (str "Exception while processing: " msg))))
+            (error e 
+                   "Exception attempting to process with " processor " for " ns ". Msg: " msg)))
         (recur (async/<! tapper))))
     (async/tap (:out cm)
                tapper)
     tapper))
 
+(defn calling-namespace
```

Comment:
this gets the namespace a few steps back up the stack. It's not *this* namespace, it's the one that called it. It allows me to have a decent error msg when something goes wrong with a selector. It's a bit dirty to be fair, but it's hard to know where anonymous functions have come from!